src/configuration.ts

```diff
@@ -111,19 +114,98 @@ export function findConfigurationPath(suppliedConfigFilePath: string, inputFileP
-        // remove BOM if present
-        fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config: IConfigFile;
+        let configFileDir: string;
+        if (fs.existsSync(configFilePath)) {
+            let fileData = fs.readFileSync(configFilePath, "utf8");
+            // remove BOM if present
+            fileData = fileData.replace(/^\uFEFF/, "");
+            config = JSON.parse(fileData) as IConfigFile;
+            configFileDir = path.dirname(configFilePath);
+        } else {
+            config = require(configFilePath);
+            configFileDir = path.resolve(`./node_modules/${configFilePath}`);
+        }
+
+        if (config.rulesDirectory != null) {
+            if (typeof config.rulesDirectory === "string") {
+                config.rulesDirectory = [<string>config.rulesDirectory];
+            }
+
+            for (let index = 0, len = config.rulesDirectory.length; index < len; index++) {
+                const directory = config.rulesDirectory[index];
+                if (directory.indexOf(".") === 0) {
+                    config.rulesDirectory[index] = getRelativePath(directory, configFileDir);
+                }
+            }
+        }
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames != null) {
+            if (typeof parentConfigNames === "string") {
+                parentConfigNames = [<string>parentConfigNames];
+            }
+
+            for (const name of parentConfigNames) {
+                const parentFilePath = resolveConfigurationPath(name, configFileDir);
+                const parentConfig = loadConfigurationFromPath(parentFilePath);
+                config = applyExtends(config, parentConfig);
+            }
+        }
+
+        return config;
     }
 }
 
+export interface IConfigFile {
+    extends?: string | string[];
+    rulesDirectory?: string | string[];
+    rules?: any;
+}
+
+/**
+ * Resolve configuration file path
+ * @var relativeFilePath Relative path or package name (tslint-config-X) or package short name (X)
+ */
+function resolveConfigurationPath(relativeFilePath: string, relativeTo: string) {
+    let filePath = relativeFilePath;
+    if (relativeFilePath.indexOf(".") !== 0 && relativeFilePath.indexOf("tslint-config") === -1) {
+        filePath = `tslint-config-${relativeFilePath}`;
+    } else if (relativeFilePath.indexOf(".") === 0) {
+        filePath = getRelativePath(relativeFilePath, relativeTo);
+    }
+
+    return filePath;
+}
+
+function applyExtends(config: IConfigFile, parentConfig: IConfigFile): IConfigFile {
+    let parentRulesDirectory = parentConfig.rulesDirectory;
+    if (parentRulesDirectory != null) {
+        if (config.rulesDirectory == null) {
+            config.rulesDirectory = parentRulesDirectory;
+        } else {
+            config.rulesDirectory = (<string[]>config.rulesDirectory).concat(<string[]>parentRulesDirectory);
```

Comment:
`config.rulesDirectory` will already be `string[]` here because of https://github.com/palantir/tslint/pull/1016/files#diff-8b01d5598b81332eb62385d26f3ead22R138

I agree that it is a bit logic bleeding, so change suggestion welcome.